### ISSUE TITLE
Offline Mutations when async is not updating the cache with the response

### ISSUE THREAD

**Issue by doldyn** [awslabs/aws-mobile-appsync-sdk-js#298]
>  Note: If your issue/feature-request/question is regarding the AWS AppSync service, please log it in the 
  [official AWS AppSync forum](https://forums.aws.amazon.com/forum.jspa?forumID=280&start=0)

**Do you want to request a *feature* or report a *bug*?**
Bug
**What is the current behavior?**
When posting a mutation and the user is online, the cache is automatically updated with the response. If the user is offline and the mutation is queued, when the user is online again, the mutation is posted to the server but the client Apollo cache is not updated with the mutation response.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
Can be reproduced with any mutation by going offline.

**What is the expected behavior?**
The client cache is updated correctly with the server response irrespective if the user is online or offline when performing the mutation.
**Which versions and which environment (browser, react-native, nodejs) / OS are affected by this issue? Did this work in previous versions?**
Tested in React.js. Do not believe it is specific for browsers. Issue exists in latest version 1.4.0 but was also present in previous releases.

**Comment by elorzafe** [comment]
@doldyn can you provide a code snippet of your mutation?

**Comment by doldyn** [comment]
Hi, thank you for looking into this issue. Today, I checked and noticed that version 1.5.1 was released yesterday. I upgraded to this latest release and the issue appears to have been resolved. I was able to systematically get the wrong state where the view wasn't updated after having been offline in the previous release. I have now tried several times after the update and have not been able to get it at all. I will close this issue and reopen with more details if it appears again. Once again, thank yo for looking into it.
